easybuild/easyconfigs/v/VASP/VASP-5.4.4-18Apr17-p01-foss-2018b.eb

```diff
@@ -0,0 +1,44 @@
+easyblock = 'MakeCp'
+
+name = 'VASP'
+version = '5.4.4-18Apr17-p01'
+
+homepage = 'http://www.vasp.at'
+description = """The Vienna Ab initio Simulation Package (VASP) is a computer program for atomic scale
+materials modelling, e.g. electronic structure calculations and quantum-mechanical molecular dynamics,
+from first principles."""
+
+toolchain = {'name': 'foss', 'version': '2018b'}
+toolchainopts = {'usempi': True}
+
+# Vasp is proprietary software, see http://www.vasp.at/index.php/faqs on how to get access to the code
+sources = ['%(namelower)s.%(version)s.tar.gz']
+
+# patch from Vaspwiki, http://cms.mpi.univie.ac.at/patches/patch.5.4.4.16052018.gz
+patches = ['vasp.5.4.4.16052018.patch']
+
+checksums = [
+    '5bd2449462386f01e575f9adf629c08cb03a13142806ffb6a71309ca4431cfb3',  # vasp.5.4.4.tar.gz
+    '3392a0ffccfa17a77775be86b1bed74f8f75a0a642f77ee0fae59950d0a3d3f4',  # vasp.5.4.4.16052018.patch
+]
+
+prebuildopts = 'cp arch/makefile.include.linux_gnu ./makefile.include && '
+
+# Makefile uses LIBS as a list of folders
+prebuildopts += 'unset LIBS && '
+
+targets = ['std', 'gam', 'ncl']
+
+buildopts = '%s FCL="$MPIF90" OFLAG="$OPTFLAGS" BLAS="$LIBBLAS" LAPACK="$LIBLAPACK" ' % ' '.join(targets)
```

Comment:
If you want the OFLAGS to really take effect you'd need something like:
```
buildopts += 'DEBUG="$FFLAGS" OFLAG="$FFLAGS" OFLAG_1="$FFLAGS" OFLAG_2="$FFLAGS" '
buildopts += 'OFLAG_3="$FFLAGS" CFLAGS_LIB="$CFLAGS" FFLAGS_LIB="$FFLAGS" '
```
And we don't set FCL so I don't really think it is required.

But I usually build our patched version which has decent makefiles...

Reply:
@akesandgren well, I suppose the vasp developers have their reasons (?) for the different optimization levels for different objects, I just wanted to redefine the default, and mostly to add the optarch flag to it (and according to the easybuild logs, that's what is happening)

But you are right that to add other flags to the other levels, we should probably use something like e.g. `OFLAG_1=${OPTFLAGS/O2/O1}`, etc. (?)

Reply:
I wouldn't trust the VASP developers to really have a reason for the different flags that is still valid.
It's basically a remnant of really old VASP 4 builds and haven't changed much that I can remember. (Well, at least not which files they compile with different flags)
And you need to look at the src/makefile to see what they really do with those _1/2/3 flags, it's not pretty.